File: wear/src/main/java/net/osdn/gokigen/joggingtimer/stopwatch/ITimerCounter.java
package net.osdn.gokigen.joggingtimer.stopwatch;

import java.util.List;

public interface ITimerCounter
{
    boolean isStarted();
    boolean isReset();
    void start();
    void stop();
    long timeStamp();
    void reset();

    int getElapsedCount();
    long getPastTime();
    long getElapsedTime(int lapCount);
    long getLastElapsedTime();
    long getCurrentElapsedTime();

    long getStartTime();
    long getStopTime();

    List<Long> getLapTimeList();
    List<Long> getReferenceLapTimeList();
    long getReferenceLapTime(int position);
    void selectReferenceLapTime(int id);

    void setCallback(MyTimerCounter.ICounterStatusNotify callback);
}

File: wear/src/main/java/net/osdn/gokigen/joggingtimer/stopwatch/timer/ITimerCounter.kt
package net.osdn.gokigen.joggingtimer.stopwatch.timer

data class LapTimeInfo(val lapTime: Long, val isPass: Boolean)

interface ITimerCounter
{
    fun isStarted(): Boolean
    fun isReset(): Boolean
    fun toggleCounterMode()
    fun getCounterMode() : Boolean
    fun getCurrentCountStatus() : ICounterStatus
    fun start()
    fun stop()
    fun timeStamp(isPass: Boolean): Long
    fun reset()
    fun getLapTimeCount(): Int
    fun getPastTime(): Long
    fun getLapTime(lapCount: Int): Long
    fun isPassLapTime(lapCount: Int): Boolean
    fun getLastLapTime(): Long
    fun getCurrentLapTime(): Long
    fun getStartTime(): Long
    fun getStopTime(): Long
    fun getLapTimeList(): List<LapTimeInfo>
    fun getReferenceLapTimeList(refId: Int): List<Long>?
    fun getReferenceLapTime(refId: Int, position: Int): Long
    fun setCallback(callback: ICounterStatusNotify)
}
